portfolio/src/main/webapp/commentsection.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function getComments() {
+    const showNum = getCommentLimit();
+    document.getElementById('shownum').value = parseInt(showNum);
+    fetch('/comment-data?shownum='+showNum).then(response => response.json()).then((comments) => {
+        const commentsContainer = document.getElementById("commentsection-container");
+        commentsContainer.innerHTML = "";
+        comments.forEach((line) => {
+            commentsContainer.appendChild(createComment(line));
+        });
+    });
+}
+
+function createComment(commentData) {
+    let res = document.createElement("div");
+    res.setAttribute("class", "single-comment");
+    
+    let username = document.createElement("div");
+    username.setAttribute("class", "username");
+    username.innerText = commentData.username + ":";
+
+    let time = document.createElement("div");
+    time.setAttribute("class", "time");
+    let val = -1;
+    let units = "";
+    if(commentData.time < 1000*60) {
+        val = Math.floor(commentData.time/1000);
+        units = "second";
+    } else if(commentData.time < 1000*60*60) {
+        val = Math.floor(commentData.time/(1000*60));
+        units = "minute";
+    } else if(commentData.time < 1000*60*60*24) {
+        val = Math.floor(commentData.time/(1000*60*60));
+        units = "hour";
+    } else if(commentData.time < 1000*60*60*24*365) {
+        val = Math.floor(commentData.time/(1000*60*60*24));
+        units = "day";
+    } else {
+        val = Math.floor(commentData.time/(1000*60*60*24));
+        units = "year";
+    }
```

Comment:
Prefer standard duration library here. Create a duration object from current time and comment time, then format that.

https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html

Reply:
I think this overcomplicates things, mostly because right now I'm doing the computations on the client side which is written in Javascript. If I used this library I'd have to perform the computations in the server and then send over the correct numerical time and the unit it's using, which can be different for each comment.